exec/java-exec/src/main/java/org/apache/drill/exec/store/pcap/PcapBatchReader.java

```diff
@@ -0,0 +1,323 @@
+
+    public PcapReaderConfig(PcapFormatPlugin plugin) {
+      this.plugin = plugin;
+    }
+  }
+
+  public PcapBatchReader(PcapReaderConfig readerConfig) {
+    this.readerConfig = readerConfig;
+  }
+
+  @Override
+  public boolean open(FileSchemaNegotiator negotiator) {
+    split = negotiator.split();
+    openFile(negotiator);
+    SchemaBuilder builder = new SchemaBuilder();
+    pcapSchema = new Schema();
+    TupleMetadata schema = pcapSchema.buildSchema(builder);
+    negotiator.setTableSchema(schema, false);
+    loader = negotiator.build();
+
+    // Creates writers for all fields (Since schema is known)
+    rowWriter = loader.writer();
+    typeWriter = rowWriter.scalar("type");
+    timestampWriter = rowWriter.scalar("packet_timestamp");
+    timestampMicroWriter = rowWriter.scalar("timestamp_micro");
+    networkWriter = rowWriter.scalar("network");
+    srcMacAddressWriter = rowWriter.scalar("src_mac_address");
+    dstMacAddressWriter = rowWriter.scalar("dst_mac_address");
+    dstIPWriter = rowWriter.scalar("dst_ip");
+    srcIPWriter = rowWriter.scalar("src_ip");
+    srcPortWriter = rowWriter.scalar("src_port");
+    dstPortWriter = rowWriter.scalar("dst_port");
+    packetLengthWriter = rowWriter.scalar("packet_length");
+
+    //Writers for TCP Packets
+    tcpSessionWriter = rowWriter.scalar("tcp_session");
+    tcpSequenceWriter = rowWriter.scalar("tcp_sequence");
+    tcpAckNumberWriter = rowWriter.scalar("tcp_ack");
+    tcpFlagsWriter = rowWriter.scalar("tcp_flags");
+    tcpParsedFlagsWriter = rowWriter.scalar("tcp_parsed_flags");
+    tcpNsWriter = rowWriter.scalar("tcp_flags_ns");
+    tcpCwrWriter = rowWriter.scalar("tcp_flags_cwr");
+    tcpEceWriter = rowWriter.scalar("tcp_flags_ece");
+    tcpFlagsEceEcnCapableWriter = rowWriter.scalar("tcp_flags_ece_ecn_capable");
+    tcpFlagsCongestionWriter = rowWriter.scalar("tcp_flags_ece_congestion_experienced");
+
+    tcpUrgWriter = rowWriter.scalar("tcp_flags_urg");
+    tcpAckWriter = rowWriter.scalar("tcp_flags_ack");
+    tcpPshWriter = rowWriter.scalar("tcp_flags_psh");
+    tcpRstWriter = rowWriter.scalar("tcp_flags_rst");
+    tcpSynWriter = rowWriter.scalar("tcp_flags_syn");
+    tcpFinWriter = rowWriter.scalar("tcp_flags_fin");
+
+    dataWriter = rowWriter.scalar("data");
+    isCorruptWriter = rowWriter.scalar("is_corrupt");
+
+    return true;
+  }
+
+  @Override
+  public boolean next() {
+    while (!rowWriter.isFull()) {
+      if (!parseNextPacket(rowWriter)) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  @Override
+  public void close() {
+    try {
+      fsStream.close();
+    } catch (IOException e) {
+      throw UserException.
+        dataReadError().addContext("Error closing InputStream: " + e.getMessage()).build(logger);
+    }
+    fsStream = null;
+    this.buffer = null;
+    this.decoder = null;
+  }
+
+  private void openFile(FileSchemaNegotiator negotiator) {
+    String filePath = null;
```

Comment:
Remove redundant variable, ti can created in try / catch block.

Reply:
Removed...